### PR TITLE
chore: swapping react to use rollup

### PR DESCRIPTION
**Pull request by JessicaSachs**
This PR adds support for ESM by switching to rollup. We're keeping the existing webpack configuration for now until we audit what example packages rely on it for a "base" configuration.

### How to test
1. cd npm/react
2. yarn build
3. yarn cypress:open

---

Cypress component tests _still use webpack to bundle the user source code_ however the `@cypress/react` library is bundled with rollup

### CODE REVIEW

File: npm/react/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "@cypress/react",
   "version": "0.0.0-development",
   "description": "Unit test React components using Cypress",
-  "main": "dist",
+  "main": "dist/cypress-react.cjs.js",
```
Comment (elevatebart) [inline npm/react/package.json:5]: should we add a module export?

**Review by lmiller1990 — APPROVED** [review]

**Review by lmiller1990 — APPROVED** [review]

**Review by elevatebart — CHANGES_REQUESTED** [review]
If I understand correctly, you switched `tsc` for `rollup`.
This means that you will have to deal with the typings separately.

What script build those typings on build?

**Comment by elevatebart** [comment]
As a clarification, are you saying that without this PR `@cypress/react` was bundled with webpack?
I think it was not bundled at all and only transpiled using TypeScript `tsc`.

**Review by elevatebart — DISMISSED** [review]
Just checked it out and it is actually generating the typings.... my bad, rollup has evolved a lot since my days.

File: npm/react/package.json [hunk 2]
```diff
@@ -2,9 +2,9 @@
...
   "scripts": {
-    "build": "rimraf dist && yarn transpile",
+    "build": "rimraf dist && yarn rollup -c rollup.config.js",
```
Comment (elevatebart) [inline npm/react/package.json:7]: Why are the watch scripts not updated?

**Review by elevatebart — APPROVED** [review]
Try this again

**Pull request merged**